### PR TITLE
More precise falsey isset()

### PR DESCRIPTION
**Pull request by staabm**
split off parts out from https://github.com/phpstan/phpstan-src/pull/2657 since it gets too complicated.

lets do it one step at a time instead

### CODE REVIEW

File: src/Analyser/TypeSpecifier.php [hunk 1]
```diff
@@ -681,14 +679,35 @@ public function specifyTypesInCondition(
...
+				foreach ($expr->vars as $var) {
+					$isset = $scope->issetCheck($var, static fn () => true);
+
+					if ($isset !== true) {
+						continue;
+					}
+
+					$type = $scope->getType($var);
+					if (
+						!$var instanceof Expr\Variable ||
```
Comment (ondrejmirtes) [inline src/Analyser/TypeSpecifier.php]: I'm not sure what's special about `Variable` that it needs to be mentioned in this condition, it's usually a red flag.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Analyser/TypeSpecifier.php (hunk 1 above)
Comment (staabm) [reply src/Analyser/TypeSpecifier.php]: Its the first step to get this better as before. 

Isset on variable is a very common case. I will make this more general in followups

Comment (ondrejmirtes) [reply src/Analyser/TypeSpecifier.php]: Please make this correct now, it's the right time to do it. The logic is correct when there is no special handling of a specific expr types.

File: src/Analyser/TypeSpecifier.php [hunk 2]
```diff
@@ -681,14 +679,34 @@ public function specifyTypesInCondition(
...
+				foreach ($expr->vars as $var) {
+					$isset = $scope->issetCheck($var, static fn () => true);
+
+					if ($isset !== true) {
+						continue;
+					}
+
+					$type = $scope->getType($var);
+					if (
+						!$type instanceof MixedType && !TypeCombinator::containsNull($type)
```
Comment (ondrejmirtes) [inline src/Analyser/TypeSpecifier.php]: `!$type instanceof MixedType && ` doesn't need to be there as `TypeCombinator::containsNull` will never return `true` for `MixedType`

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Analyser/TypeSpecifier.php (hunk 2 above)
Comment (staabm) [reply src/Analyser/TypeSpecifier.php]: without this explicit `MixedType` check the 

```
function mixedIsset(mixed $m): void
{
	if (isset($m)) {
		assertType("mixed~null", $m);
	} else {
		assertType("null", $m);
	}
}
```

case will not succeed. I turned the conditions arround, maybe its more readable now.

**Comment by ondrejmirtes** [comment]
I knew deep down the condition really wasn't needed. `instanceof` of a specific type is also usually a red flag: https://github.com/phpstan/phpstan-src/pull/2708/commits/001ad45ab5f11aba6baffba512aad56f9ccc740a

**Comment by ondrejmirtes** [comment]
Thank you.

**Comment by staabm** [comment]
awesome, thanks!

**Pull request merged**